gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/ExternalDataConfiguration.java

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Google BigQuery configuration for tables backed by external data. Objects of this class describe
+ * the data format, location, and other properties of a table stored outside of BigQuery.
+ * By defining these properties, the data source can then be queried as if it were a standard
+ * BigQuery table. Support for external tables is experimental and might be subject to changes or
+ * removed.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/federated-data-sources">Federated Data Sources
+ *     </a>
+ */
+public class ExternalDataConfiguration implements Serializable {
```

Comment:
Yes, a load job has its own configuration (which is slightly different). This class is also used to define [external data sources to a query job](https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.query.tableDefinitions), which are query-scoped federated tables (described [here](https://cloud.google.com/bigquery/federated-data-sources#cli))